### PR TITLE
Issue #68: Stats page liquidity data 

### LINKED ISSUE #68: Stats page liquidity data
depends on https://github.com/UseKeyp/od-app/issues/60
Depends on https://github.com/UseKeyp/od-sdk/issues/6

- [ ] Fetch pool data for OD and ODG from camelot. Include this data in the existing hooks for fetching data
- [ ] Use the SDK to set values for the stats page 
     - `marketPriceODG`
     - `marketPriceOD`
     - `ODETHUniswapLiquidity`

image

- [ ] Use the SDK to set OD liquidity values for the header bar. 
     - Market price OD
     - liquidity in pool

image

- [ ] Update header bar liquidity dropdown icon from uniswap to camelot

### PR DESCRIPTION
**Pull request by jahabeebs**
Closes #68 

## Description
- Fetching market price of OD and total pool liquidity--displaying on stats page, navbar/header component, side menu component
- Modified navbar/header and side menu component to display the accurate market price of OD as well as the liquidity of the pool according to the figma designs
- Fixed bug that prevented closing navbar/header and side menu components when clicking on them directly once open (now can close them by clicking on them when open or by clicking outside of them)

Note that currently the designs include a "view on camelot exchange" link which I've included in the navbar/header and side menu components. do we want to leave this in and I'll put in the link later or leave it out?

Note that we can't merge this until fetchPoolData is included in the latest SDK release

## Screenshots

Desktop

https://github.com/open-dollar/od-app/assets/47253537/0c808617-ac29-4ff9-8b09-b111e0fbda09

Mobile

https://github.com/open-dollar/od-app/assets/47253537/00e9cca3-04b6-4eaa-acb4-b72f96ca7887

Analytics page

prices

### CODE REVIEW

**Review by pi0neerpat — APPROVED** [review]

**Comment by pi0neerpat** [comment]
Dollar-format the total liquidity

**Comment by jahabeebs** [comment]
@pi0neerpat I added the fetchPoolData function to the pool and whatever I do it's always undefined, yet the data is being fetched correctly (state.poolFetchedData = payload in connectWalletModel.ts is correctly being set, but whenever the pool data is accessed it is always undefined). I tried fetching the pool data using the store both in Shared.tsx and in the analytics page (by doing ```await connectWalletActions.fetchPoolData(geb)```). Can you think of why this might be? Otherwise I'm going to just fetch the data from the SDK function

File: src/containers/Shared.tsx [hunk 1]
```diff
@@ -166,6 +167,7 @@ const Shared = ({ children, ...rest }: Props) => {
     }, [auctionsData, setInternalBalance, setProtInternalBalance])
 
     useEffect(() => {
+        connectWalletActions.setPoolFetchedData(poolData)
         connectWalletActions.setTokensData(tokensData)
     }, [connectWalletActions, tokensData])
```
Comment (pi0neerpat) [inline src/containers/Shared.tsx:172]: Maybe poolData needs to be added here `[connectWalletActions, tokensData, poolData]`

Comment (jahabeebs) [reply src/containers/Shared.tsx]: I actually didn't end up using poolData at all--it wasn't being used. I tried adding poolData as an async function being fetched in Shared as well (latest push) but still not working

**Comment by pi0neerpat** [comment]
My understanding was that OD/ETH Liquidity in Camelot needed the pool data.

Lets forget the store issue and move on. If this one is ready I can review

image

**Comment by jahabeebs** [comment]
fixed price formatting issue and branch is now in line with dev so merging

fixed price

**Pull request merged**